Src/AutoFixture/DataAnnotations/AttributeRelay.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.DataAnnotations
+{
+    /// <summary>
+    /// Relays requests made for properties, fields or parameters to different requests, based on custom attributes 
+    /// applied to the code elements. 
+    /// </summary>
+    /// <remarks>
+    /// A typical example is relaying a request for a parameter that has <see cref="RangeAttribute"/> applied to it 
+    /// to a <see cref="RangedNumberRequest"/>. Another way to describe it, a request for a <see cref="ParameterInfo"/>
+    /// instance that contains a <see cref="RangeAttribute"/>, is transformed to a <see cref="RangedNumberRequest"/>
+    /// with parameters extracted from the attribute.
+    /// </remarks>
+    /// <typeparam name="TAttribute">
+    /// An <see cref="Attribute"/> type that can be applied to a property, a field or a method. The attribute is 
+    /// expected to allow applying only once. When the same attribute is applied multiple times to the same code 
+    /// element, an <see cref="InvalidOperationException"/> will be thrown.
+    /// </typeparam>
+    public class AttributeRelay<TAttribute> : ISpecimenBuilder

```

Comment:
For this particular set of changes, the `AttributeRelay<TAttribute>` could be `internal`. However, to help with implementation of attribute relays outside of the `Ploeh.AutoFixture` assembly it would need to be either public or `[InternalsVisibleTo]`. In particular, I'd like to reuse it to implement the `SubstituteAttribute` in the `Ploeh.AutoFixture.NSubstitute` assembly.
from the `Ploeh.AutoFixture.NSubstitute` assembly. (This was initially discussed in #132 and previously submitted in #421)